packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/util/iterateLinkedObjects.ts

```diff
@@ -43,8 +43,11 @@ export async function* iterateLinkedObjects<T extends OntologyObject>(
       },
     );
 
+    const { targetType } =
+      client.ontology.objects[sourceApiName].links[linkTypeApiName];
```

Comment:
Actual bugfix - use the correct targetType instead of the link type when we convert the linked objects